### ISSUE TITLE
v0.1.8 errors everytime with The storage key for descriptor 'acs.NDc3...' could not be found.

### ISSUE THREAD

**Issue by drdamour** [microsoft/terraform-provider-azuredevops#492]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

❯ tf -v
Terraform v1.0.11
on windows_amd64
+ provider registry.terraform.io/microsoft/azuredevops v0.1.8

### Affected Resource(s)

* `azuredevops_users`

### Terraform Configuration Files

```hcl
data "azuredevops_users" "all" {
}
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by xuzhang3** [comment]
@drdamour   This root issue is  the enhance PR #482 to export user ID.  By default the `getStoragekey` operation will loop all the found users to get the user ID. In your configuration, this will loop all the users in your organization.   On workaround is narrow the search condition to reduce the user list which will speed up the operation.  
In my opinion, to fix this:
1. Disable this feature and add a new resource to export user ID based on customer usage scenario
2. Add a flag to enable/disable export user ID for this resource.

Do you have any  other suggestions,?

**Comment by drdamour** [comment]
> @drdamour This root issue is the enhance PR #482 to export user ID. By default the `getStoragekey` operation will loop all the found users to get the user ID. In your configuration, this will loop all the users in your organization. On workaround is narrow the search condition to reduce the user list which will speed up the operation. In my opinion, to fix this:
> 
> 1. Disable this feature and add a new resource to export user ID based on customer usage scenario
> 2. Add a flag to enable/disable export user ID for this resource.
> 
> Do you have any other suggestions,?

no, i'd prefer #1 i think, though i can understand why people would want #2 to avoid a foreach...but it's sooooo slow.

**Comment by bgmonroe** [comment]
Coming here >1 year later and I agree with @drdamour.  Ideally there would be two data sources: a singular version accepting a single email/UPN identifier, returning a singleton object and a plural version accepting a list of email/UPN identifiers and returning a list of objects.  I know MS doesn't maintain the AzureAD Terraform provider but I would recommend (for consistency) looking to it for inspiration (see: [azuread_user](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/data-sources/user)/[azuread_users](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/data-sources/users) and [azuread_group](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/data-sources/group)/[azuread_groups](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/data-sources/groups) data sources).
